TCP_UDP_server.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"log"
+	"bytes"
+	"fmt"
+	"io"
+	"sync"
+	"net"
+)
+var routinecount sync.WaitGroup
+var waitForgoroutine sync.WaitGroup
+var message string
+
+func main()  {
+	waitForgoroutine.Add(1)
+	go deployTcpServer()
+	deployUdpserver()
+	waitForgoroutine.Wait()
+
+}
+
+func deployTcpServer()  {
+
+	defer waitForgoroutine.Done()
+	// listens for connection
+	listens,error:=net.Listen("tcp","localhost:5002")
+	fmt.Println("Server is listening: ",listens.Addr())
+	if error!=nil{
+		log.Fatal(error)
+	}
+	// listens for connection
+	for{
+		connection,error:=listens.Accept()
+		if error!=nil {
+			log.Fatal(error)
+			continue
+		}
+		fmt.Println(connection.RemoteAddr()," connected to TCP server")
+
+
+
+
+
+		go replyTCPClient(connection)
+
+	}
+
+
+}
+
+
+
+func replyTCPClient(con net.Conn)  {
+	defer con.Close()
+
+	// read is used to read the data from connection stream
+	//we are making a buffer to read data from client for this we have a buffer which read byte by byte
+	buf:=make([]byte,1024)
+	_, err := con.Read(buf)
+	if err!=nil{
+		log.Fatal(err)
+
+	}
+
+	n := bytes.Index(buf, []byte{0})
+	mesage:=string(buf[:n-1]) //
+	fmt.Println(con.RemoteAddr()," says: ",mesage)
+	// writing reply to client
+	_,error1:=io.WriteString(con,"Hey i am server you said: "+mesage)
```

Comment:
Why not to use `con.Write()`?

Reply:
1: you told me there are several method to read and write message so i was trying if could know about more methods :) 
i was little confused about using variable with same name again and again so i did it in this way

Reply:
But you are violating the rule reusability. I think only thing that should matter is the scope of variable and the need of value inside that variable.

Reply:
yeah i am getting your point i will make an func so that i can use it again and again